### PR TITLE
Update NSW_OEH pipeline to handle single-beam data

### PR DESCRIPTION
**Pull request by mhidas**
This will require the corresponding harvester update: https://github.com/aodn/harvesters/pull/529
and config updates https://github.com/aodn/chef-private/pull/2377, https://github.com/aodn/chef/pull/538

### CODE REVIEW

**Comment by mhidas** [comment]
@lbesnard @lbesnard Anyone up for a quick review?

**Comment by mhidas** [comment]
Could someone please have a brief look at this and merge if possible? I'd like to test this in RC this week. It might be easier to make sense of each commit separately.

File: AODN/NSW-OEH/process_zip.py [hunk 1]
```diff
@@ -81,269 +84,353 @@ def get_survey_name(path):
...
+        # open zip file and read content list
+        if not zipfile.is_zipfile(self.zip_file):
+            report[self.zip_file] = ["Not a valid zip archive!"]
+            return report
+        with zipfile.ZipFile(self.zip_file) as zf:
+            path_list = zf.namelist()
+
+        # Check each individual file name
+        have_metadata = False
+        have_coverage = False
```
Comment (pmbohm) [inline AODN/NSW-OEH/process_zip.py]: Why no use an integer and increment? That will get rid of one TODO further down

**Comment by bpasquer** [comment]
All good for me 👍  @mhidas

**Comment by mhidas** [comment]
Thanks @bpasquer @pmbohm ! I'll merge it now so I can test on RC.

**Comment by mhidas** [comment]
@pblain Could we post-review this please? Thanks

**Comment by pblain** [comment]
@mhidas - sure

File: AODN/NSW-OEH/process_zip.py (hunk 1 above)
Comment (mhidas) [inline AODN/NSW-OEH/process_zip.py]: Thanks, good idea. I will update in my next PR.

File: AODN/NSW-OEH/test_process.py [hunk 2]
```diff
@@ -2,128 +2,246 @@
...
+
+    def test_check_name_short(self):
+        pz = self.proc('NSWOEH_20151029_PortHackingBateBay_MB.zip')
+        self.assertEqual(["File name should have at least 4 underscore-separated fields."],
+                         pz.check_name('NSWOEH_20151029_PortHackingBateBay.zip')
+                         )
+
+    def test_check_name_bad_method(self):
+        pz = self.proc('NSWOEH_20151029_PortHackingBateBay_MB.zip')
+        self.assertEqual(["Field 4 should be a valid survey method code"],
```
[the remark this answers is not in the document]
Comment (mhidas) [reply AODN/NSW-OEH/test_process.py]: Thanks @anguss00 . I'll take a look at refactoring this the next time I need to make a change to the code. However, I found there's a trade-off with looping through test data: you need to explicitly print out which item it failed on. For non-looped calls like here that's taken care of by the default output from `unittest`. So if it's only a couple of repeats, I'm not sure it's worth it.

**Pull request merged**